package.json

```diff
@@ -225,19 +226,25 @@
     "cover-integration": "cross-env COVERAGE=true nyc --reporter=lcovonly yarn mocha \"test/integration/**/*.test.[tj]s\" && node -e \"require('fs').renameSync('coverage/lcov.info', 'coverage/integration.info')\"",
     "cover-unit": "cross-env COVERAGE=true nyc --reporter=lcovonly yarn mocha \"test/unit/**/*.test.[tj]s\" && node -e \"require('fs').renameSync('coverage/lcov.info', 'coverage/unit.info')\"",
     "merge-coverage": "lcov-result-merger \"coverage/*.info\" \"coverage/lcov.info\"",
-    "----------------------------------------- local test dbs ------------------------------------------": "",
-    "start-mariadb": "bash dev/mariadb/10.3/start.sh",
-    "start-mysql": "bash dev/mysql/5.7/start.sh",
-    "start-mysql-8": "bash dev/mysql/8.0/start.sh",
-    "start-postgres": "bash dev/postgres/10/start.sh",
-    "start-mssql": "bash dev/mssql/2019/start.sh",
-    "start-db2": "bash dev/db2/11.5/start.sh",
-    "stop-mariadb": "bash dev/mariadb/10.3/stop.sh",
-    "stop-mysql": "bash dev/mysql/5.7/stop.sh",
-    "stop-mysql-8": "bash dev/mysql/8.0/stop.sh",
-    "stop-postgres": "bash dev/postgres/10/stop.sh",
-    "stop-mssql": "bash dev/mssql/2019/stop.sh",
-    "stop-db2": "bash dev/db2/11.5/stop.sh",
+    "----------------------------------------- bulk tests ------------------------------------------": "",
+    "start-all": "ts-node dev/start-all.ts",
+    "test-all": "ts-node dev/test-all.ts",
+    "stop-all": "ts-node dev/stop-all.ts",
+    "----------------------------------------- database containers -------------------------------------": "",
```

Comment:
See what I did in #14587 and see if we can add that for other dialects as well

Reply:
> but if they're following semver, each major version includes breaking changes. So each major version should be tested.

Which is why we're testing against latest (https://github.com/sequelize/sequelize/pull/14587#issuecomment-1150355327 for more information)

> `*-<specific version>`

I would rather avoid adding `*-<specific version>` to our dev folder. Issues with these versions are going to be rare enough that the added maintenance in keeping our dev folders up to date is not going to be worth it. Having the CI warn every now and then if there is an issue, and manually booting a new container is sufficient

> There's so much convenience in not having to type the extra qualifier suffix 

Personally that's just `yarn sta<tab>` followed by selecting whichever one I want using arrow keys (thanks to https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/yarn):
![image](https://user-images.githubusercontent.com/1280915/172715248-b7961826-3044-43b9-9f15-de217767523c.png)

> BTW we also have to keep in mind that `latest` and `oldest` are obfuscated.

True, that's something I considered in #14587, but ultimately could not come up with a better short name. Ideas welcome though!

Reply:
Yeah, I think Zoé mentioned `minimum` and `maximum` but I disliked that so I went with `oldest` and `latest`. What we can do to improve is mention on the releases page which versions are checked by our CI so people know that the support of postgres 14 (in this example) is not tested

Reply:
RE: @ephys 
---
> Which is why we're testing against latest (https://github.com/sequelize/sequelize/pull/14587#issuecomment-1150355327 for more information)
Yes, but I may not be explaining the problem well enough. The intermediary versions are also major versions.  _Latest_ may work, _Oldest_ may work, but there might be an intermediary major version that doesn't work.  

Consider version 6, 7, 8, which are all breaking changes.  7 (or a minor version) could have removed a feature. 8 might have added it back.  If you never test 7, there's no way of knowing.

> I would rather avoid adding `*-<specific version>` to our dev folder. Issues with these versions are going to be rare enough that the added maintenance in keeping our dev folders up to date is not going to be worth it.

The dev folder has already been out-of-date. I realized this when I saw the version number from `tree`. I don't think burying version numbers in files is going to make that issue any better.  It's your project, a more robust scheme would allow you to test against minor versions.  There's going to be time where 9.5 works and 9.6 doesn't (as with PostgreSQL).  You'll want an environment that accommodates testing/developing for those supported versions and ensure regression tests don't fail.

Keep the felxibility that the object can grown by version numbers -- specific versions in the directory structure help logging, parsing, and creating other scripts.  Create symbolic links to the versions (or implement dynamic lookups) if you don't want to maintain package.json.

I submit that the metadata of having the version numbers as directories is more beneficial than discarding them for cosmetics.

```
dev/dialects/
├── db2
│   ├── 11.5
│   │   ├── Docker
│   │   ├── start.sh
│   │   └── stop.sh
│   ├── latest -> 11.5
│   ├── oldest -> 11.5
│   └── README.md
├── mssql
│   ├── 2019
│   │   ├── docker-compose.yml
│   │   ├── start.sh
│   │   └── stop.sh
│   ├── 2022
│   │   ├── docker-compose.yml
│   │   ├── start.sh
│   │   └── stop.sh
│   ├── latest -> 2019
│   └── oldest -> 2022
```

>True, that's something I considered in https://github.com/sequelize/sequelize/pull/14587, but ultimately could not come up with a better short name. Ideas welcome though!

Unless you want to get more specific (with a qualifier), I think what you all have come up with are great! I'm not sure I'd change it.

As food for thought.  If you wanted to get more specific, it could be `mssql-latest-installed`, but that's somewhat redundant because they're all installed.  The only purpose is to serve as an overt reminder that to make you think there might be a version not installed.  At some point you might want to test against up-and-coming databases.  Something like `mssql-edge`.

@WikiRik
---
> Yeah, I think Zoé mentioned minimum and maximum but I disliked that so I went with oldest and latest. 
Minimum vs Maximum are more technically correct, but I think oldest and latest have better UX (or DX) and that'd be my preference too if I had a vote.

> What we can do to improve is mention on the releases page which versions are checked by our CI so people know that the support of postgres 14 (in this example) is not tested
I like that idea, especially if you can automate it. It's not so much about needing to test or develop for everything, but building the transparency.  If someone reports an issue where something is broken a certain version, though, that's where the flexibility of minor versions comes in.  You can do some of that natively, but why when you have such a good dev-environment process?  Lean on that, don't resist the version scheme 😆

Reply:
> Latest may work, Oldest may work, but there might be an intermediary major version that doesn't work.
>
> Consider version 6, 7, 8, which are all breaking changes. 7 (or a minor version) could have removed a feature. 8 might have added it back.

This is what I mentioned in my other comment. The probability of this happening is incredibly low. Databases are huge software designed slowly and carefully, breaking changes aren't done last minute to then realize it was a bad idea and revert it next release.

> If you never test 7, there's no way of knowing.

Again, this is planned: A once a week CI run on a broader range of databases.

> The dev folder has already been out-of-date. I realized this when I saw the version number from tree. I don't think burying version numbers in files is going to make that issue any better.

It's out of date because it's already a lot of work to keep them up to date, which is the whole reason behind the change to oldest/latest.

- `oldest` doesn't need to change at all until the next major.
- `latest` can be kept up to date automatically thanks to renovate, something we couldn't do before because version numbers were hardcoded everywhere.

Adding intermediary versions to the `dev` folder, to the test config, brings back the issue of having to regularly update everything, which won't be done because it's a lot of work for very little benefit.

An important note that may influence your opinion: The `dev` folder is not used on the CI. It's for local development. The only time you'll need an intermediary version locally is when the once-a-week CI run tells us something is broken in that version. You can then easily boot a temporary intermediary instance of the database.

> discarding them for cosmetics.

Not cosmetics, maintenance work that will not be done because the time necessary to keep it up to date is not worth the almost nil benefit it brings

> At some point you might want to test against up-and-coming databases.

That's already the case, `mssql-latest` is the upcoming 2022 version